### PR TITLE
[EIP-712] Restruct pending requests

### PR DESCRIPTION
**Pull request by pilagod**
* Restruct `pendingTransaction` to `pendingRequests`
* Will add a new request type `EIP712` in the upcoming PR.

### CODE REVIEW

**Review by oneleo — APPROVED** [review]
- Summary:
     1. Renamed `pendingTransaction` to `pendingRequests`.
     2. Changed `pendingRequests` type in Storage from `struct {id: TransactionPending}` to `array [TransactionRequest]` and added `RequestType`.

- Old `pendingTransaction` Example:

<details>
  <summary>( Please click here to expand )</summary>
    <pre><code>

  "pendingTransaction": {
    "4ba41c3a-6b67-4a99-ab38-f22724297cdc": {
      "id": "4ba41c3a-6b67-4a99-ab38-f22724297cdc",
      "status": "Pending",
      "createdAt": 1723113938,
      "senderId": "d1fe8079-99b2-4cbe-9ae1-ebb1aaca3d8d",
      "networkId": "9fcc077e-c5ce-4935-b453-0d2aee3f5d2e",
      "to": "0x1Bb8f2BCD02D04d3e2bd9816952E74BA067Ab5Be",
      "value": "0x5af3107a4000",
      "data": "0x"
    }
  }
    </code></pre>
</details>

- New `pendingRequests` Example:
<details>
  <summary>( Please click here to expand )</summary>
    <pre><code>

  "pendingRequests": [
    {
      "type": "Transaction",
      "id": "6209e243-1d4d-46aa-a9d9-20803fab8898",
      "createdAt": 1723114025,
      "senderId": "403beebb-c03e-4872-a42e-9e352045bad0",
      "networkId": "868f8f44-6372-4ff3-ac3b-b40e7095a70d",
      "to": "0x1Bb8f2BCD02D04d3e2bd9816952E74BA067Ab5Be",
      "value": "0x5af3107a4000",
      "data": "0x"
    }
  ]
    </code></pre>
</details>

File: app/storage/slice/transaction.ts [hunk 1]
```diff
@@ -86,7 +87,10 @@ export const createTransactionSlice: BackgroundStateCreator<
 
   markERC4337TransactionRejected: async (txId, data) => {
     await set(({ state, getERC4337TransactionType }) => {
-      const tx = state.pendingTransaction[txId]
+      const txIndex = state.pendingRequests.findIndex(
+        (r) => r.type === RequestType.Transaction && r.id === txId
+      )
```
Comment (WIN-LIN) [inline app/storage/slice/transaction.ts:92]: `txIndex` could be not found here, too.

**Review by WIN-LIN — APPROVED** [review]

File: app/storage/slice/transaction.ts (hunk 1 above)
Comment (pilagod) [reply app/storage/slice/transaction.ts]: I decided to throw error when not found at this moment. And I also extracted a `StateActor` to collect common logic for storage state across app and background, see [f486087](https://github.com/pilagod/waallet-extension/pull/199/commits/f486087239d461c46fee0f7cce3f486f9e150019)

Comment (pilagod) [reply app/storage/slice/transaction.ts]: Just found the typo after push 🤦‍♂️ [c53f13e](https://github.com/pilagod/waallet-extension/pull/199/commits/c53f13e994bdd889b79aa69164ef352e2a6680f2)

Comment (WIN-LIN) [reply app/storage/slice/transaction.ts]: > I decided to throw error when not found at this moment. And I also extracted a StateActor to collect common logic for storage state across app and background, see [f486087](https://github.com/pilagod/waallet-extension/pull/199/commits/f486087239d461c46fee0f7cce3f486f9e150019)

: I'm wondering what diff between `StateActor` and manager like `NetworkStorageManager` or `AccountStorageManager`) 🧐?

`StateActor` seems to dealing with transaction request  so far. Are there any actions going to be implemented?

**Pull request merged**